src/js/actions/type.js

```diff
@@ -114,6 +136,32 @@ define(function (require, exports) {
     };
     setPostScript.reads = [locks.JS_APP, locks.JS_TOOL];
     setPostScript.writes = [locks.PS_DOC, locks.JS_DOC, locks.PS_APP, locks.JS_POLICY];
+    setPostScript.modal = true;
+
+    /**
+     * Update the type face (in terms of a type family and type style) of the given
+     * layers in the given document. 
+     *
+     * @param {Document} document
+     * @param {Immutable.Iterable.<Layers>} layers
+     * @param {string} family The type face family name, e.g., "Helvetica Neue"
+     * @param {string} style The type face style name, e.g., "Oblique"
+     * @return {Promise}
+     */
+    var updateFace = function (document, layers, family, style) {
```

Comment:
Oh, I see you're actually calling the aciton methods directly... that's even more dangerous. Action queue binds things and keeps order. You definitely should transfer to these methods from `set` methods.